client/my-sites/themes/atomic-transfer-dialog.tsx

```diff
@@ -0,0 +1,160 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { Dialog } from '@automattic/components';
+import { localize, translate } from 'i18n-calypso';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import EligibilityWarnings from 'calypso/blocks/eligibility-warnings';
+import Notice from 'calypso/components/notice';
+import { getSiteSlug } from 'calypso/state/sites/selectors';
+import {
+	acceptAtomicTransferDialog,
+	dismissAtomicTransferDialog,
+	activate as activateTheme,
+	initiateThemeTransfer,
+} from 'calypso/state/themes/actions';
+import {
+	getCanonicalTheme,
+	getThemeForAtomicTransferDialog,
+	isExternallyManagedTheme,
+	shouldShowAtomicTransferDialog,
+} from 'calypso/state/themes/selectors';
+import {
+	isTransferComplete,
+	isUploadInProgress,
+} from 'calypso/state/themes/upload-theme/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { Theme } from 'calypso/types';
+import './atomic-transfer-dialog.scss';
+
+interface AtomicTransferDialogProps {
+	siteId?: number;
+	inProgress?: boolean;
+	isTransferred?: boolean;
+	showEligibility: boolean;
+	theme: Theme;
+	siteSlug?: string | null;
+	isMarketplaceProduct?: boolean;
+	dispatchAcceptAtomicTransferDialog: typeof acceptAtomicTransferDialog;
+	dispatchDismissAtomicTransferDialog: typeof dismissAtomicTransferDialog;
+	dispatchActivateTheme: typeof activateTheme;
+	dispatchRecordTracksEvent: typeof recordTracksEvent;
+	dispatchInitiateThemeTransfer: typeof initiateThemeTransfer;
+}
+
+class AtomicTransferDialog extends Component< AtomicTransferDialogProps > {
+	handleAccept() {
+		const { siteId, dispatchInitiateThemeTransfer } = this.props;
+		if ( ! siteId ) {
+			return;
+		}
+
+		dispatchInitiateThemeTransfer( siteId, null, '' );
+	}
+
+	getCurrentPath = () => {
+		const { origin, href } = window.location;
+
+		return href.replace( origin, '' );
+	};
+
+	componentDidUpdate( prevProps: Readonly< AtomicTransferDialogProps > ): void {
+		const { siteId, siteSlug, isTransferred } = this.props;
+		if ( siteId && siteSlug && prevProps.isTransferred !== isTransferred && isTransferred ) {
+			window.location.replace(
+				this.getCurrentPath().replace( /\b.wordpress.com/, '.wpcomstaging.com' )
```

Comment:
While it doesn't block shipping this patch, I think we should also try to trigger a state update. In my case, the navigation went ahead, but the new page didn't detect that I had the marketplace theme active, so the main CTA was still `Activate this design` rather than `Customize`.
Might we be seeing an issue where the site transfer has completed, but the theme hasn't been activated yet? We might need to wait for both to be true, not just that the transfer completed.